### PR TITLE
[Benchmark] Make sure decode PCC comparison uses same input as golden

### LINKED ISSUE #4614: Benchmark decode PCC compounds prefill error
The benchmark llm tester runs prefill->decode on cpu then prefill->decode on device and compares the inputs. If prefill run on device doesn't output the same last token (1st prediction) as cpu then the decode steps are running with different inputs. This can cause the decode pcc to be recorded as very low (ex `0.07`) despite seeing a high pcc when running with `decode_only=True`.

Kimi-k2 sees this happen, likely because it's a 2 layer test with a lower prefill pcc (~`0.92`). `test_kimi_k2_tp_galaxy_2_layers` fails here [To be added: [running here](https://github.com/tenstorrent/tt-xla/actions/runs/25676680580)] with a decode pcc of [To be added: locally saw `0.07`]. Running the same test with `decode_only=True` [passes here](https://github.com/tenstorrent/tt-xla/actions/runs/25673958331/job/75370138802#step:15:334) with `First decode PCC verification passed with PCC=0.987921`.

A comparison of the last prefill tokens between cpu and device (run locally) confirmed the prefill last token mismatch is the issue with only 21.9% of the tokens matching out of the batch of 64 inputs. 
```
Prefill PCC verification passed with PCC=0.917409
[DEBUG] Last-prefill-token argmax agreement: 21.9% (50/64 disagree)
```

### PR DESCRIPTION
**Pull request by odjuricicTT**
### Ticket
Fixes #4614

### Problem description
The LLM benchmark runs prefill→decode on CPU and on device, then computes a decode PCC by comparing the device first-decode logits to the CPU reference. When device prefill diverges from CPU prefill (which happens at lower-than-perfect prefill PCC), the device's first-decode input differs from the reference's first-decode input, and the decode PCC collapses (e.g. kimi-k2 saw decode PCC of ~0.07 despite the decode itself being healthy at 0.98 in `decode_only` mode).

### What's changed
- Split the device PCC/TOPK run into prefill (1 step) and decode (`logits_steps` steps).
- When device prefill output diverges from CPU prefill output, override the first-decode input with CPU's prefill output so decode PCC measures decode quality instead of compounding the prefill mismatch.
- `decode_only` mode bypasses the comparison entirely (no device prefill in that path).
- Teacher-forced ground-truth window for the decode call shifts by 1 since the prefill call already consumed `gt[0]`.

### CI run
Performance Benchmark run for this branch (all HW, non-shared runners, llms only):
https://github.com/tenstorrent/tt-xla/actions/runs/25792410589

### Checklist
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by odjuricicTT** [comment]
## Performance benchmark vs latest nightly

- **This PR run:** [25792410589](https://github.com/tenstorrent/tt-xla/actions/runs/25792410589)
- **Baseline (nightly 2026-05-12):** [25705461027](https://github.com/tenstorrent/tt-xla/actions/runs/25705461027)
- Filter applied: `test-filter=llama,phi,gemma,falcon,qwen`, `runs-on-filter=All`, non-shared runners
- Comparison generated with [`gh-perf-report`](https://github.com/rpavlovicTT/gh-perf-report)

### Pass / fail counts
| | This PR | Baseline (filter-matched) |
| --- | --- | --- |
| Passed | **40 / 50** | 17 / 24 |
| Failed | 10 / 50 | 7 / 24 |

(`NEW` rows in the gh-perf-report output: 19 — all p150 jobs that didn't exist in baseline. `REMOVED`: 23 — mostly non-LLM and out-of-filter models.)

### Status regressions (success → failure)
| Job | Baseline | This PR |
| --- | --- | --- |
| `qwen_3_32b_tp` (n300-llmbox) | ✅ | ❌ |
| `qwen_3_4b` (n150-perf) | ✅ | ❌ |

### Status improvements (failure → success)
| Job | Baseline | This PR |
| --- | --- | --- |
| `qwen_2_5_7b_instruct` (n150-perf) | ❌ | ✅ |
| `tiiuae_falcon3-1b-base` (n150-perf) | ❌ | ✅ |

### Performance regression (still passing, slower)
- `test_llama_3_1_70b_tp` (n300-llmbox): **−41.3% samples/sec** (−3.21). This PR only touches the PCC/TOPK benchmark path (no change to the perf measurement loop), so this is most likely a flaky measurement (cf #4632); worth a rerun to confirm.

### Pre-existing failures (carry over from baseline)
- `falcon3_10b_tp`, `falcon3_7b_tp` (n300-llmbox)
- `microsoft_phi-1_5`, `microsoft_phi-2`, `tiiuae_falcon3-7b-base` (n150-perf)

### p150-only failures (no baseline coverage for p150)
- `microsoft_phi-1_5`, `microsoft_phi-2`, `qwen_2_5_3b_instruct`

**Review by rpavlovicTT — APPROVED** [review]
